src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -842,8 +842,18 @@ def _build_preflight_error_message(preflight_error):
     return '\n'.join(err_messages)
 
 
+def _prepare_template_uri_with_query_string(template_uri, input_query_string):
+    from six.moves.urllib.parse import urlencode, parse_qs, urlsplit, urlunsplit  # pylint: disable=import-error
+
+    scheme, netloc, path, query_string, fragment = urlsplit(template_uri)  # pylint: disable=unused-variable
+    query_params = parse_qs(input_query_string)
+    new_query_string = urlencode(query_params, doseq=True)
```

Comment:
If there is an exception in parsing, does the current error raised meet the expectation? Do we need to consider the logic of adding `try-catch`?

Reply:
I added the try catch for cases I may not have thought of yet but the current errors raised does meet expectations. If the query string is not valid then query_params ={} and is not url encoded and the server side handles any errors very well. 

I did notice that when the url is retrieved to get the template_obj and prompt for parameters, if the template-uri is incorrect then the error message returned is <urlopen error [Errno 11001] getaddrinfo failed> for the current Azure CLI release. I did not make any changes to this.

Reply:
> <urlopen error[Errno 11001]getaddrinfo failed>

This error message doesn't seem clear. Should we consider catching the parsing logic of `template_uri`?

Reply:
This actually comes from def _urlretrieve(url):